File: backend/eCommerce-Project/src/main/java/com/eCommerce/eCommerce/Project/dto/Purchase.java
package com.eCommerce.eCommerce.Project.dto;

import com.eCommerce.eCommerce.Project.entity.Address;
import com.eCommerce.eCommerce.Project.entity.Customer;
import com.eCommerce.eCommerce.Project.entity.Order;
import com.eCommerce.eCommerce.Project.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: frontend/angular-ecommerce/src/app/common/purchase.ts
import { OrderItem } from './order-item';
import { Order } from './order';
import { Address } from './address';
import { Customer } from './customer';
export class Purchase {
  cusomter: Customer;
  shippingAddress: Address;
  billingAddress: Address;
  order: Order;
  orderItem: OrderItem[];
}
